compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1212,7 +1212,9 @@ public void visit(BLangForever foreverStatement) {
         isSiddhiRuntimeEnabled = foreverStatement.isSiddhiRuntimeEnabled();
         foreverStatement.setEnv(env);
         for (StreamingQueryStatementNode streamingQueryStatement : foreverStatement.getStreamingQueryStatements()) {
-            analyzeStmt((BLangStatement) streamingQueryStatement, env);
+            SymbolEnv stmtEnv = SymbolEnv.createStatementEnv((BLangStatement) streamingQueryStatement, env);
+            resolveStreamingScopeEntries(stmtEnv, streamingQueryStatement);
```

Comment:
Better to move this line to `visit(BLangStreamingQueryStatement streamingQueryStatement) {}`? 

So we simply create the env and call `analyzeNode()` for each queryStmt, and its' visit() method will handle anything specific to the query statement.